component/alerts.libsonnet

```diff
@@ -0,0 +1,31 @@
+local thanosMixin = import 'github.com/thanos-io/thanos/mixin/mixin.libsonnet';
+local thanos = import 'kube-thanos/thanos.libsonnet';
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.thanos;
+
+local ruleEnabled = function(enabledRules, rule)
+  (std.objectHas(enabledRules, '*') && enabledRules['*'] == true)
+  || (std.objectHas(enabledRules, rule.alert) && enabledRules[rule.alert] == true);
+
+local fromMixin =
+  function(name, mixinGroupName, enabledRules) kube._Object('monitoring.coreos.com/v1', 'PrometheusRule', name) {
+    metadata+: {
+      namespace: params.namespace,
+    },
+    spec+: {
+      groups+:
+        std.filterMap(
+          function(group) group.name == mixinGroupName,
+          function(group) group { rules: std.filter(function(rule) ruleEnabled(enabledRules, rule), super.rules) },
+          thanosMixin.prometheusAlerts.groups
+        ),
+    },
+  };
```

Comment:
> Shouldn't we also ensure that alerts are working with our current setup on OCP4, cf. https://github.com/projectsyn/component-rook-ceph/blob/26104e0d5d73fc6effd6005eb4f2bc52fca23ab7/component/alertrules.libsonnet#L86-L127 and the rest of that file for an example (note that this logic still should be moved to somewhere central).

I did not know about that. I'll take a look.

> Additionally, it would be nice if we can adjust useful, but badly configured alerts through the hierarchy if necessary (local rulepatch = com.getValueOrDefault(patch_alerts, rule.alert, {}); in the linked file)

Good idea!

Reply:
I hope i did understand correctly to make the alerts work:
- Added the syn labels
- Allowed patches